utils/browser_history.jsx

```diff
@@ -3,4 +3,41 @@
 
 import {createBrowserHistory} from 'history';
 
-export const browserHistory = createBrowserHistory({basename: window.basename});
+import {isServerVersionGreaterThanOrEqualTo} from 'utils/server_version';
+import {isDesktopApp, getDesktopVersion} from 'utils/user_agent';
+
+const b = createBrowserHistory({basename: window.basename});
+const isDesktop = isDesktopApp() && isServerVersionGreaterThanOrEqualTo(getDesktopVersion(), '5.0.0');
+
+window.addEventListener('message', ({origin, data: {type, message = {}} = {}} = {}) => {
+    if (origin !== window.location.origin) {
+        return;
+    }
+
+    switch (type) {
+    case 'browser-history-push-return': {
+        const {pathName} = message;
+        b.push(pathName);
+        break;
+    }
+    }
+});
+
+export const browserHistory = {
+    ...b,
```

Comment:
Is the object created by `createBrowserHistory` a plain object or is it one that actually has a class and methods? I think this might be fine given how weird classes are in JS, but I'm trying to think if we should be extending a class or something here.

Reply:
Looks like it's just a POJO: https://github.com/ReactTraining/history/blob/28c89f4091ae9e1b0001341ea60c629674e83627/packages/history/index.ts#L559

I think either way we're good to just copy over the contents and override the methods we want to override, normally I'd be in favour of inheritance but it doesn't look like that's possible here.